Foundation/NSString.swift

```diff
@@ -1331,8 +1331,13 @@ open class NSMutableString : NSString {
         }
 
         let start = _storage.utf16.startIndex
-        let min = _storage.utf16.index(start, offsetBy: range.location).samePosition(in: _storage)!
-        let max = _storage.utf16.index(start, offsetBy: range.location + range.length).samePosition(in: _storage)!
+        let min = _storage.utf16.index(start, offsetBy: range.location).samePosition(in: _storage) ??
+            _storage.utf16.index(start, offsetBy: _storage.rangeOfComposedCharacterSequence(at: range.location).location)
+
+        let max = _storage.utf16.index(start, offsetBy: range.location + range.length).samePosition(in: _storage) ?? {
```

Comment:
Looks reasonable, though from a style perspective it might be nice to make these consistent (either both as a closure or not — I prefer the closure).

In the cases where we might expect to fall into the `rangeOfComposedCharacterSequence(at:)` case for both `min` and `max`, it's likely worth factoring out that computation, I think — there are cases where it's not so cheap.

Reply:
Am I correct that the computation for `rangeOfComposedCharacterSequence(at:)` can only be reused if `range.length == 0`. Is that a valid input?

Reply:
🤦‍♂️
You're totally right, of course — read too hastily.